sr_health_check/src/sr_health_check/sr_health_check_gui.py

```diff
@@ -0,0 +1,418 @@
+        '''
+            Saves the results to the result file.
+            @param results: results to be saved
+        '''
+        if results:
+            self._current_data.update(results)
+            with open(self._results_file, 'w', encoding="ASCII") as yaml_file:
+                yaml.safe_dump(self._current_data, stream=yaml_file, default_flow_style=False)
+
+    def update_passed_label(self, check, text=None):
+        '''
+            Updates the passed label in the GUI with feedback on completion status.
+            @param check: Check object
+            @param text: String stating the state of the check.
+        '''
+        if isinstance(check, MotorCheck):
+            self._widget.passed_motor.setText(text or str(self._checks_to_execute['motor']
+                                              ['check'].has_passed()))
+        elif isinstance(check, PositionSensorNoiseCheck):
+            self._widget.passed_position_sensor.setText(text or str(self._checks_to_execute['position_sensor_noise']
+                                                        ['check'].has_passed()))
+        elif isinstance(check, MonotonicityCheck):
+            self._widget.passed_monotonicity.setText(text or str(self._checks_to_execute['monotonicity']
+                                                     ['check'].has_passed()))
+        elif isinstance(check, TactileCheck):
+            self._widget.passed_tactile.setText(text or str(self._checks_to_execute['tactile']
+                                                ['check'].has_passed()))
+        elif isinstance(check, BacklashCheck):
+            self._widget.passed_backlash.setText(text or str(self._checks_to_execute['backlash']
+                                                 ['check'].has_passed()))
+        elif isinstance(check, OverrunCheck):
+            self._widget.passed_overrun.setText(text or str(self._checks_to_execute['overrun']
+                                                ['check'].has_passed()))
+
+    def combobox_selected(self):
+        '''
+            Performs action upon date combobox selection from View tab
+        '''
+        self.display_data()
+
+    def display_data(self):
+        '''
+            Updates the results tree to be presented.
+        '''
+        if not self._current_data:
+            return
+
+        for data_entry in list(self._current_data.keys()):
+            if self._widget.combobox_date.findText(data_entry) == -1:  # findText returns the index of of the item or -1
+                self._widget.combobox_date.insertItem(0, data_entry)
+
+        self._widget.treeWidget.clear()
+        date = self._widget.combobox_date.currentText()
+        if date != '':
+            self.update_tree(self._current_data[date], None)
+
+    def update_tree(self, data, parent):  # pylint: disable=R1710
+        '''
+            Creates the results tree to be presented.
+            @param data: dictionary containing the data
```

Comment:
Does data need to be strictly a dictionary? There is an if that checks if data is a dict, but it will also do something if data is "something else". Perhaps it would be nice to specify what that something else should be in the documentation

Reply:
There is no documentation for the return either

Reply:
Agreed, and maybe `return None` rather than overriding the pylint.

Reply:
I'll check if pylint won't complain about returning different types. Regarding the dictionary part - this is supposed to be a recursive method, the initial argument is the dictionary with the test results which on the lowest level is a string value passed as an argument. Not sure how to change the pydoc for this scenario.